### ISSUE TITLE
Release a version for GHC 9.0.1

### ISSUE THREAD

**Issue by andreasabel** [ocharles/weeder#65]
The latest version on hackage does not want to build with GHC 9.0:
```
$ cabal install weeder -w ghc-9.0.1
Resolving dependencies...
cabal: Could not resolve dependencies:
[__0] trying: weeder-2.1.3 (user goal)
[__1] next goal: base (dependency of weeder)
[__1] rejecting: base-4.15.0.0/installed-4.15.0.0 (conflict: weeder =>
base^>=4.13.0.0 || ^>=4.14.0.0)
```
(I suspect there is more effort required to move to GHC 9.0 than a base bump.)

**Comment by ocharles** [comment]
Thanks for raising this @andreasabel. I don't currently have a GHC 9 compiler, but I'll try and get this sorted (I'm mostly waiting for haskell.nix)

**Comment by mimi1vx** [comment]
@ocharles isn't 9.0.1 in nix for a pretty long time? as an optional compiler?

**Comment by ocharles** [comment]
@mimi1vx I don't use `haskellPackages`, but use `haskell.nix`. Fortunately, GHC 9 is now available there, so I think I can start looking into this soon!

**Comment by ocharles** [comment]
Unfortunately we're blocked on basement for now

**Comment by mimi1vx** [comment]
yes, and hackage head has patch for basement up to ghc 9.2 https://gitlab.haskell.org/ghc/head.hackage/-/blob/master/patches/basement-0.0.11.patch

**Comment by LeventErkok** [comment]
@ocharles 

Just a note about GHC 9.0.1 users; this is quite confusing. If you have GHC 9.0.1, and If you simply issue:

```
 cabal new-install weeder
```

you get weeder 1.0.9 installed; as I guess that version had compatible bounds with base 4.15. And if you aren't paying attention to the version cabal installed (which I wasn't!) you end up being quite confused about why you get a bizarre behavior. (See now closed #76).

Would be really nice to have a GHC 9 series release indeed.
